### PR TITLE
Garbage collection of VPA checkpoints

### PR DESCRIPTION
**Pull request by schylek**

### CODE REVIEW

File: vertical-pod-autoscaler/recommender/input/cluster_feeder.go [hunk 1]
```diff
@@ -159,6 +165,35 @@ func (feeder *clusterStateFeeder) InitFromCheckpoints() {
...
+		namespace := namespaceItem.Name
+		checkpointList, err := feeder.vpaCheckpointClient.VerticalPodAutoscalerCheckpoints(namespace).List(metav1.ListOptions{})
+		if err != nil {
+			glog.Errorf("Cannot list VPA checkpoints from namespace %v. Reason: %+v", namespace, err)
+		}
+		for _, checkpoint := range checkpointList.Items {
+			vpaID := model.VpaID{Namespace: checkpoint.Namespace, VpaName: checkpoint.Spec.VPAObjectName}
+			_, exists := feeder.clusterState.Vpas[vpaID]
+			if !exists {
+				err = feeder.vpaCheckpointClient.VerticalPodAutoscalerCheckpoints(namespace).Delete(checkpoint.Name, &metav1.DeleteOptions{})
```
Comment (mwielgus) [inline vertical-pod-autoscaler/recommender/input/cluster_feeder.go]: Add some logs on the deleted resources.

File: vertical-pod-autoscaler/recommender/input/cluster_feeder.go [hunk 2]
```diff
@@ -159,6 +165,37 @@ func (feeder *clusterStateFeeder) InitFromCheckpoints() {
...
+				if err == nil {
+					glog.Errorf("VPA checkpoint %v/%v deleted.", namespace, checkpoint.Name)
```
Comment (mwielgus) [inline vertical-pod-autoscaler/recommender/input/cluster_feeder.go]: This is not an error.

Comment (schylek) [reply vertical-pod-autoscaler/recommender/input/cluster_feeder.go]: Fixed.

File: vertical-pod-autoscaler/recommender/input/cluster_feeder.go [hunk 3]
```diff
@@ -159,6 +165,37 @@ func (feeder *clusterStateFeeder) InitFromCheckpoints() {
...
+		if err != nil {
+			glog.Errorf("Cannot list VPA checkpoints from namespace %v. Reason: %+v", namespace, err)
+		}
+		for _, checkpoint := range checkpointList.Items {
+			vpaID := model.VpaID{Namespace: checkpoint.Namespace, VpaName: checkpoint.Spec.VPAObjectName}
+			_, exists := feeder.clusterState.Vpas[vpaID]
+			if !exists {
+				err = feeder.vpaCheckpointClient.VerticalPodAutoscalerCheckpoints(namespace).Delete(checkpoint.Name, &metav1.DeleteOptions{})
+				if err == nil {
+					glog.V(4).Infof("VPA checkpoint %v/%v deleted.", namespace, checkpoint.Name)
```
[the remark this answers is not in the document]
Comment (schylek) [reply vertical-pod-autoscaler/recommender/input/cluster_feeder.go]: Changed to V(3) - is there any policy on log levels for k8s we I could refer to?
Info that this is a cleanup added.

**Pull request merged**